### PR TITLE
Newsletters: add 285 (2024-01-17)

### PR DESCRIPTION
**Pull request by harding**
- [x] Lede, releases/RCs, topic entries @harding 
- [x] Bitcoin Core 29058 @adamjonas 
- [x] Other merges @harding

Includes Ark topic that was previously in #1452

### CODE REVIEW

File: _posts/en/newsletters/2024-01-17-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+- **Disclosure of past vulnerability in Core Lightning:** Matt Morehouse
+  used Delving Bitcoin to [announce][morehouse delving] a vulnerability
+  he had previously [responsibly disclosed][topic responsible
+  disclosures] that affected Core Lightning versions 23.02 through
+  23.05.2.  More recent versions of 23.08 or higher are not affected.
+
+    The new vulnerability was discovered by Morehouse's following up on
+    his prior work on fake funding, which he also responsibly disclosed
+    (see [Newsletter #266][news266 lnbugs]).  When re-testing nodes that
+    had implemented fixes for fake funding, was able to trigger a [race
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-01-17-newsletter.md:23]:
```suggestion
    had implemented fixes for fake funding, he was able to trigger a [race
```

File: _posts/en/newsletters/2024-01-17-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+  an output, so they could directly include an internal key in a script.
+  However, `OP_INTERNALKEY` uses an old suggestion <!-- FIXME:cite -->
+  from CTV's original author, Jeremy Rubin, to save several vbytes and
+  potentially make scripts more reusable by allowing the value of the
+  key to be retrieved from the script interpreter.
+
+    In the thread, Black and others describe some of the protocols that
+    would be enabled by this combination of consensus changes:
+    [LN-Symmetry][topic eltoo] (eltoo), [Ark][topic ark]-style
+    [joinpools][topic joinpools], reduced-signatures [DLCs][topic dlc],
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-01-17-newsletter.md:53]: Maybe? 

```suggestion
    [joinpools][topic joinpools], reduced-signature [DLCs][topic dlc],
```

File: _posts/en/newsletters/2024-01-17-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+    For more information, we recommend reading Morehouse's excellent
+    [full disclosure][morehouse full] blog post.
+
+- **New LNHANCE combination soft fork proposed:** Brandon Black
+  [posted][black lnhance] to Delving Bitcoin details about a soft fork
+  that combines previous proposals for [OP_CHECKTEMPLATEVERIFY][topic
+  op_checktemplateverify] (CTV) and [OP_CHECKSIGFROMSTACK][topic
+  op_checksigfromstack] (CSFS) with a new proposal for an
+  `OP_INTERNALKEY` that places the taproot internal key on the stack.
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-01-17-newsletter.md:42]: We have a good explanation of taproot internal keys here: https://bitcoinops.org/en/newsletters/2019/05/14/#complex-spending-with-taproot

Perhaps that or another resource would be helpful for reader

File: _posts/en/newsletters/2024-01-17-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+  [posted][daftuar cluster] a summary of the [cluster mempool][topic
+  cluster mempool] proposal to Delving Bitcoin.  Optech attempted to
+  summarize the current state of cluster mempool discussion in
+  [Newsletter #280][news280 cluster] but we would strongly recommend
+  reading the overview by Daftuar, who is one of the architects of the
+  proposal.  One detail we have not previously covered caught our
+  attention:
+
+    - *CPFP carve out needs to be removed:* the [CPFP carve out][topic
+      cpfp carve out] mempool policy added to Bitcoin Core in FIXME:year
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-01-17-newsletter.md:87]: 2019? https://bitcoinops.org/en/newsletters/2019/07/24/#bitcoin-core-15681

File: _posts/en/newsletters/2024-01-17-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+      transactions to delay consideration of a child transaction
+      belonging to an honest peer.  The carve out allows one transaction
+      to slightly exceed the limits.  In cluster mempool, related
+      transactions are placed in a cluster and the limits are applied
+      per cluster, not per transaction.  Under that policy, there's no
+      known way to ensure a cluster only contains a maximum of one carve
+      out unless we restrict the relationships allowed between
+      transactions relayed on the network far beyond the current
+      restrictions.  A cluster with multiple carve outs could
+      significantly exceed its limits, at which point we'd need to
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-01-17-newsletter.md:100]: Maybe reword to take out "we’d"

File: _posts/en/newsletters/2024-01-17-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+      engineer the protocol for those much higher limits. That would
+      accommodate users of carve outs but restrict what regular
+      transaction broadcasters can do---an undesireable proposition.
+
+      A proposed solution to the incompatibility between carve out and
+      cluster mempool is [v3 transaction relay][topic v3 transaction
+      relay], which would allow regular users of v1 and v2 transactions
+      to continuing using them in all the historically typical ways, but
+      also allow the users of contract protocols like LN to opt-in to v3
```
Comment (vostrnad) [inline _posts/en/newsletters/2024-01-17-newsletter.md:109]:
```suggestion
      also allow the users of contract protocols like LN to opt in to v3
```

File: _posts/en/newsletters/2024-01-17-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+  original proposal.  Briar describes the notable changes: "removing the
+  grinding of the nLocktime in favor of a relative block height, which
+  all of the compressed inputs use, and the use of a second kind of
+  variable integer."
+
+- **Discussion of Miner Extractable Value (MEV) in non-zero ephemeral anchors:**
+  Gregory Sanders [posted][sanders mev] to Delving Bitcoin to discuss
+  concerns about [ephemeral anchor][topic ephemeral anchors] outputs
+  that contain more than 0 satoshis.  An ephemeral anchor pays an output
+  script equivalent to `OP_TRUE`, allowing anyone to spend it.
```
Comment (vostrnad) [inline _posts/en/newsletters/2024-01-17-newsletter.md:142]:
```suggestion
  that contain more than 0 satoshis.  An ephemeral anchor pays to an output
  script equivalent to `OP_TRUE`, allowing anyone to spend it.
```
Personally I'd slightly rewrite this:
```suggestion
  that contain more than 0 satoshis.  An ephemeral anchor pays to a
  standardized output script that anyone can spend.
```

File: _posts/en/newsletters/2024-01-17-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+  potentially make scripts more reusable by allowing the value of the
+  key to be retrieved from the script interpreter.
+
+    In the thread, Black and others describe some of the protocols that
+    would be enabled by this combination of consensus changes:
+    [LN-Symmetry][topic eltoo] (eltoo), [Ark][topic ark]-style
+    [joinpools][topic joinpools], reduced-signatures [DLCs][topic dlc],
+    and [vaults][topic vaults] without presigned transactions, among
+    other described benefits of the underlying proposals, such as
+    CTV-style congestion control and CSFS-style signature delegation.
```
Comment (murchandamus) [inline _posts/en/newsletters/2024-01-17-newsletter.md:56]: Ugh, congestion control is such an absurd application, I really don’t understand why people keep bringing that up.

**Review by murchandamus — COMMENTED** [review]
Very cool newsletter so far.

File: _posts/en/newsletters/2024-01-17-newsletter.md (hunk 8 above)
Comment (harding) [reply _posts/en/newsletters/2024-01-17-newsletter.md:56]: It's still the first thing described in BIP119:

> This BIP proposes a new opcode, OP_CHECKTEMPLATEVERIFY, to be activated as a change to the semantics of OP_NOP4.
>
> The new opcode has applications for transaction congestion control and [...]

(But I also tend to think that it's impractical.)

File: _posts/en/newsletters/2024-01-17-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,305 @@
...
+
+  No clear conclusion seemed to have been reached at the time of writing.
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [LND 0.0.119][] in a new release of this library for building
```
Comment (vostrnad) [inline _posts/en/newsletters/2024-01-17-newsletter.md:206]:
```suggestion
- [LDK 0.0.119][] in a new release of this library for building
```

File: _posts/en/newsletters/2024-01-17-newsletter.md [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+
+  No clear conclusion seemed to have been reached at the time of writing.
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [LND 0.0.119][] is a new release of this library for building
```
Comment (azuchi) [inline _posts/en/newsletters/2024-01-17-newsletter.md:206]:
```suggestion
- [LDK 0.0.119][] is a new release of this library for building
```

File: _posts/en/newsletters/2024-01-17-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+[stewart 64]: https://delvingbitcoin.org/t/64-bit-arithmetic-soft-fork/397/
+[daftuar cluster]: https://delvingbitcoin.org/t/an-overview-of-the-cluster-mempool-proposal/393/
+[sanders mev]: https://delvingbitcoin.org/t/ephemeral-anchors-and-mev/383/
+[bip 64]: https://github.com/bitcoin/bips/pull/1538
+[taproot internal key]: /en/newsletters/2019/05/14/#complex-spending-with-taproot
+[news56 carveout]: /en/newsletters/2019/07/24/#bitcoin-core-15681
+[news269 rpc]: /en/newsletters/2023/09/20/#bitcoin-core-28448
+[news85 stuck]: /en/newsletters/2020/02/19/#c-lightning-3500
+[news89 stuck]: /en/newsletters/2020/03/18/#eclair-1319
+[lnd 0.0.119]: https://github.com/lightningdevkit/rust-lightning/releases/tag/v0.0.119
```
Comment (azuchi) [inline _posts/en/newsletters/2024-01-17-newsletter.md:310]:
```suggestion
[ldk 0.0.119]: https://github.com/lightningdevkit/rust-lightning/releases/tag/v0.0.119
```

File: _posts/en/newsletters/2024-01-17-newsletter.md (hunk 10 above)
Comment (harding) [reply _posts/en/newsletters/2024-01-17-newsletter.md:206]: Oof, thank you @azuchi !

File: _posts/en/newsletters/2024-01-17-newsletter.md (hunk 9 above)
Comment (harding) [reply _posts/en/newsletters/2024-01-17-newsletter.md]: Thank you @vostrnad!  I just realized I missed the correction from LND to LDK.

**Comment by harding** [comment]
Made all edits (thanks @bitschmidty @vostrnad @murchandamus @azuchi !), added lede and releases, reviewed @adamjonas contribution (thanks!), and added topic entries.  Thanks everyone!

**Pull request merged**